app/controllers/events/view/tickets/attendees/list.js

```diff
@@ -37,19 +37,29 @@ export default Controller.extend({
   ],
   actions: {
     toggleCheckIn(attendee) {
-      attendee.toggleProperty('isCheckedIn');
       if (attendee.isCheckedIn) {
         let newCheckinTimes = attendee.get('checkinTimes') === null ? `${moment().toISOString()}` : `${attendee.get('checkinTimes')},${moment().toISOString()}`;
         attendee.set('checkinTimes', newCheckinTimes);
+        attendee.toggleProperty('isCheckedIn');
+        attendee.save()
+          .then(() => {
+            this.notify.success(this.get('l10n').t('Attendee Checked-Out Successfully'));
+            this.send('refreshRoute');
+          })
+          .catch(() => {
+            this.notify.error(this.get('l10n').t('An unexpected error has occurred'));
+          });
+      } else {
+        attendee.toggleProperty('isCheckedIn');
```

Comment:
what's the point of putting this statement in both if and else blocks. Write it once outside both if it is executed irrespective of the condition being true or false.

Reply:
If the toggle property i would outside the if , else block and then i need to put attendee.save outside too as the functionality will create error and if i put .save() outside if, else  , Build error would occur due to .then and .catch .
That is why i repeated the .toggleProperty in both if , else .
@CosmicCoder96

Reply:
> If the toggle property I would outside the if 
I don't know what that means.
>  then i need to put attendee.save outside too
Why would you need to put it "outside too"

Reply:
@CosmicCoder96 Did the changes